src/main/java/com/sap/cloudfoundry/client/facade/adapters/CloudFoundryClientFactory.java

```diff
@@ -1,25 +1,47 @@
 package com.sap.cloudfoundry.client.facade.adapters;
 
 import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.text.MessageFormat;
 import java.time.Duration;
 import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
 import java.util.function.Function;
 
+import com.sap.cloudfoundry.client.facade.util.CloudUtil;
 import org.cloudfoundry.client.CloudFoundryClient;
+import org.cloudfoundry.client.v3.organizations.OrganizationsV3;
+import org.cloudfoundry.client.v3.spaces.SpacesV3;
 import org.cloudfoundry.reactor.ConnectionContext;
 import org.cloudfoundry.reactor.DefaultConnectionContext;
 import org.cloudfoundry.reactor.client.ReactorCloudFoundryClient;
+import org.cloudfoundry.reactor.client.v3.organizations.ReactorOrganizationsV3;
+import org.cloudfoundry.reactor.client.v3.spaces.ReactorSpacesV3;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.immutables.value.Value;
 
+import com.sap.cloudfoundry.client.facade.rest.CloudSpaceClient;
 import com.sap.cloudfoundry.client.facade.oauth2.OAuthClient;
+import com.sap.cloudfoundry.client.facade.util.JsonUtil;
 
-import reactor.netty.http.client.HttpClient;
+import reactor.core.publisher.Mono;
 
 @Value.Immutable
 public abstract class CloudFoundryClientFactory {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(CloudFoundryClientFactory.class);
+
+    static final HttpClient HTTP_CLIENT = HttpClient.newBuilder()
+                                                    .executor(Executors.newSingleThreadExecutor())
```

Comment:
This does mean that every request/response will be handled by one thread right?

Reply:
Yes, it should be.